dist/calendar/CalendarDate.js

```diff
@@ -78,23 +78,29 @@ var CalendarDate = _react2['default'].createClass({
       mouseDown: false
     };
   },
+  componentWillUnmount: function componentWillUnmount() {
+    this.isUnmounted = true;
+  },
 
   mouseUp: function mouseUp() {
     this.props.onSelectDate(this.props.date);
-
-    if (this.state.mouseDown) {
-      this.setState({
-        mouseDown: false
-      });
+    if (!this.isUnmounted) {
+      if (this.state.mouseDown) {
+        this.setState({
+          mouseDown: false
+        });
+      }
+      document.removeEventListener('mouseup', this.mouseUp);
     }
-    document.removeEventListener('mouseup', this.mouseUp);
   },
 
   mouseDown: function mouseDown() {
-    this.setState({
-      mouseDown: true
-    });
-    document.addEventListener('mouseup', this.mouseUp);
+    if (!this.isUnmounted) {
```

Comment:
Do we really dynamic events binding/unbinding? The following seems to work (although I didn't test in many browsers):

#### 1. Instead of using state to store *mouseDown*, we can just use an instance property

The state property *mouseDown* does not affect any part of the presentation (UI), so I don't see why it should reside there.

```js
React.createClass({
  isMouseDown: false,

  mouseLeave() {
    if (this.isMouseDown) {
      // ...
    }
  }
});
```

#### 2. Events can be attached directly to the TD element, so we won't have to worry about unmounting

```js
// ...
onTouchEnd={this.touchEnd}
onMouseUp={this.onMouseUp}
// ...
```